conan/tools/meson/layout.py

```diff
@@ -1,4 +1,8 @@
+import os
+
+
 def meson_layout(conanfile):
     conanfile.folders.build = "build-{}".format(str(conanfile.settings.build_type).lower())
+    conanfile.folders.generators = os.path.join(conanfile.folders.build, "conan")
```

Comment:
Missing this was hiding in test the fact that shared libs doesn't work, because dll was in "lib" folder, not in "bin"